### PR TITLE
issue #1650: Adding detailed doxgen comments for CCDictionary.

### PR DESCRIPTION
**Pull request by dumganhar**

### CODE REVIEW

**Comment by dumganhar** [comment]
I also make the constructor and destructor of CCDictElement as private method
This will be more precise since `CCDictElement` can't be allocated except within CCDictionary.
Because CCDictionary is its friend class.

**Comment by dumganhar** [comment]
I think destructor of `CCDictElement` should not be private.
I will update it later.

File: cocos2dx/cocoa/CCDictionary.h [hunk 1]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
+     *  For internal use, invoked by setObject.
+     */
     void setObjectUnSafe(CCObject* pObject, const std::string& key);
     void setObjectUnSafe(CCObject* pObject, const int key);
     
 public:
+    /**
+     *  All the elements in dictionary.
+     * 
+     *  @note For internal use, we need to make this member variable as public since it's used in UT_HASH.
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h]: make -> declare ?

File: cocos2dx/cocoa/CCDictionary.h [hunk 2]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
+     */
     CCDictElement* m_pElements;
 private:
     
+    /** The support type of dictionary, it's be confirmed when once setObject is invoked. */
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h]: it's be confirmed when once setObject is invoked -> it's confirmed when setObject is invoked

File: cocos2dx/cocoa/CCDictionary.h [hunk 3]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
+     *        It means that when you don't need it, CC_SAFE_RELEASE needs to be invoked.
+     *
+     *  @param  pFileName  The name of the plist file.
+     *  @return A dictionary which isn't an autorelease object.
+     */
     static CCDictionary* createWithContentsOfFileThreadSafe(const char *pFileName);
 
 private:
+    /** 
+     *  For internal use, invoked by setObject.
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h]: use -> usage

File: cocos2dx/cocoa/CCDictionary.h [hunk 4]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
 
-    /*
-    @brief The same meaning as dictionaryWithContentsOfFile(), but it doesn't call autorelease, so the
-           invoker should call release().
-    */
+    /**
+     *  Create a dictionary with a plist file.
+     *  
+     *  @note the return object isn't an autorelease object.
+     *        This can make sure not using autorelease pool in different thread.
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h]: different -> a new?

File: cocos2dx/cocoa/CCDictionary.h [hunk 5]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
+     *  Create a dictionary.
+     *  @return A dictionary which is an autorelease object.
+     *  @see createWithDictionary(CCDictionary*), createWithContentsOfFile(const char*), createWithContentsOfFileThreadSafe(const char*).
+     */
     static CCDictionary* create();
 
+    /**
+     *  Create a dictionary with an exist dictionary.
+     *
+     *  @param srcDict The exist dictionary.
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h:351]: exist -> existing

File: cocos2dx/cocoa/CCDictionary.h [hunk 6]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
+     *  Remove an object by the specified integer key.
+     *
+     *  @param key  The integer key for searching.
+     *  @see removeObjectForKey(const std::string&), removeObjectsForKeys(CCArray*),
+     *       removeObjectForElememt(CCDictElement*), removeAllObjects().
+     */
     void removeObjectForKey(int key);
+    
+    /**
+     *  Remove an object by an array.
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h]: Remove objects by an array of keys.

File: cocos2dx/cocoa/CCDictionary.h [hunk 7]
```diff
@@ -39,142 +39,360 @@ of this software and associated documentation files (the "Software"), to deal
...
+    /** The support type of dictionary, it's be confirmed when once setObject is invoked. */
     enum CCDictType
     {
         kCCDictUnknown = 0,
         kCCDictStr,
         kCCDictInt
     };
+    
+    /** 
+     *  The type of dictionary, it's assigned with kCCDictUnknown by default.
```
Comment (minggo) [inline cocos2dx/cocoa/CCDictionary.h]: with -> to ?

**Pull request merged**